### PR TITLE
Reposition test

### PR DESCRIPTION
**Pull request by InderKhera**
- created reposition function to flip img, rotate img, and shift img on x and y axis
- created tests for function with 100% coverage
- added input_checker into function after we have finalized sanity checker

### CODE REVIEW

File: tests/test_reposition_image.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+@pytest.mark.parametrize("flip, rotate, shift_x, shift_y, expected", [
+    ('none', 'up', 0, 0, np.array([[1, 2], [3, 4]])),  # No transformation
+    ('horizontal', 'down', 0, 0, np.array([[3, 4], [1, 2]])),  # Horizontal flip, down rotation
+    ('vertical', 'left', 0, 0, np.array([[4, 2], [3, 1]])),  # Vertical flip, left rotation
+    ('both', 'right', 0, 0, np.array([[2, 4], [1, 3]])),  # Both flips, right rotation
+])
+def test_erroneous_cases_that_pass(flip, rotate, shift_x, shift_y, expected):
+    img = np.array([[1, 2], [3, 4]])
+    result = reposition_image(img, flip=flip, rotate=rotate, shift_x=shift_x, shift_y=shift_y)
+    assert np.array_equal(result, expected), f"Unexpected result for flip={flip}, rotate={rotate}, shift_x={shift_x}, shift_y={shift_y}"
```
Comment (JennyonOort) [inline tests/test_reposition_image.py:32]: I am having a bit trouble understanding the name of this part of test functions.

It seems this is a test case that passed but it is called an error case. 

Did I miss something here?

**Comment by InderKhera** [comment]
You're right to point out the confusion! That is a typo on my part, from revising my earlier functions and making amendments. I can rename the test to something like test_valid_transformations to clarify that these are valid inputs that should pass successfully. Thanks for catching that!

**Review by JennyonOort — CHANGES_REQUESTED** [review]
Do let me know if any issues or anything not clear from the comments!

File: tests/test_reposition_image.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    with pytest.raises(TypeError, match="shift_x must be an integer."):
+        reposition_image(img, shift_x='a')
+
+# Test invalid shift_y parameter
+def test_invalid_shift_y():
+    img = np.array([[1, 2], [3, 4]])
+    with pytest.raises(TypeError, match="shift_y must be an integer."):
+        reposition_image(img, shift_y='b')
+
+# Test invalid image type
```
[the remark this answers is not in the document]
Comment (InderKhera) [reply tests/test_reposition_image.py:72]: This function is checking if the shift_y argument is an integer or not I dont believe that would have been specified in the input_checker. The shift_y argument in the reposition_image() moves the image on its y axis

File: tests/test_reposition_image.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    with pytest.raises(ValueError, match="rotate must be one of 'up', 'left', 'right', or 'down'."):
+        reposition_image(img, rotate='invalid')
+
+# Test invalid shift_x parameter
+def test_invalid_shift_x():
+    img = np.array([[1, 2], [3, 4]])
+    with pytest.raises(TypeError, match="shift_x must be an integer."):
+        reposition_image(img, shift_x='a')
+
+# Test invalid shift_y parameter
```
[the remark this answers is not in the document]
Comment (InderKhera) [reply tests/test_reposition_image.py:66]: same as my previous comment but for the x-axis

File: tests/test_reposition_image.py (hunk 1 above)
Comment (InderKhera) [reply tests/test_reposition_image.py:32]: You're right to point out the confusion! That is a typo on my part, from revising my earlier functions and making amendments. I can rename the test to something like test_valid_transformations to clarify that these are valid inputs that should pass successfully. Thanks for catching that!

File: tests/test_reposition_image.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    ('horizontal', 'down', 0, 0, np.array([[3, 4], [1, 2]])),  # Horizontal flip, down rotation
+    ('vertical', 'left', 0, 0, np.array([[4, 2], [3, 1]])),  # Vertical flip, left rotation
+    ('both', 'right', 0, 0, np.array([[2, 4], [1, 3]])),  # Both flips, right rotation
+])
+def test_erroneous_cases_that_pass(flip, rotate, shift_x, shift_y, expected):
+    img = np.array([[1, 2], [3, 4]])
+    result = reposition_image(img, flip=flip, rotate=rotate, shift_x=shift_x, shift_y=shift_y)
+    assert np.array_equal(result, expected), f"Unexpected result for flip={flip}, rotate={rotate}, shift_x={shift_x}, shift_y={shift_y}"
+
+# Test edge cases
```
[the remark this answers is not in the document]
Comment (InderKhera) [reply tests/test_reposition_image.py:34]: Hi Jenny I have added warning like you recommended!

File: tests/test_reposition_image.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (InderKhera) [reply tests/test_reposition_image.py:66]: I have amended both tests to account for the float test instead of a string test

File: tests/test_reposition_image.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JennyonOort) [reply tests/test_reposition_image.py:101]: oops, I think it was commented to the wrong place => meant for the line # test invalid image type.
